src/domUpdates.js

```diff
@@ -289,30 +292,42 @@ const populateRecipeHeader = currentRecipe => {
   `
 }
 
-const openRecipeCard = () => {
-  allRecipes.classList.add('blur')
-  body.classList.add('no-scroll')
-  clickedRecipe.classList.toggle("hidden");
-  clickedRecipe.classList.toggle("flex");
-  clickedRecipe.classList.toggle("fade-in");
-}
-
 const showRecipe = (recipeCard) => {
   updateCurrentRecipe(recipeCard);
   populateRecipeHeader(pageData.currentRecipeCard);
   populateInstructions(pageData.currentRecipeCard);
   populateIngredients(pageData.currentRecipeCard);
   updateSaveButtons(pageData.currentRecipeCard.id, modalAddBtn, modalRemoveBtn, currentUser);
-  openRecipeCard();
+  openInfoPanel(recipeCard);
   patchHits(pageData.currentRecipeCard)
 };
 
-const closeRecipe = () => {
+const openInfoPanel = (infoType) => {
+  let thisPanel;
+  if (infoType.id === 'settings') {
+    thisPanel = settingsPanel;
+    thisPanel.classList.toggle('hidden');
+    new Promise(makeRecipeClickChart);
```

Comment:
Out of curiosity, why do we need to make a new promise with the chart? Is there anything to do with this promise/resolve it?

Reply:
It's possible we might not need to, but in the examples I was reading many were making it async to give time for the data to populate.  Lemme see what happens if I remove the promise.

Reply:
yup, still works without the promise - removing.